packages/netlify-cms-lib-util/src/API.ts

```diff
@@ -139,100 +139,107 @@ type Backend = "github" | "gitlab" | "bitbucket"
 // - `params` property for customizing response
 // - `backend`(compulsory) to specify which backend to be used: Github, Gitlab etc.
 
-type RequestConfig = Omit<RequestInit, 'headers'> & HeaderConfig & {
-  backend: Backend,
-  params?: ParamObject,
-}
+type RequestConfig = Omit<RequestInit, 'headers'> &
+  HeaderConfig & {
+    backend: Backend;
+    params?: ParamObject;
+  };
 
 export const rootApi = {
   github: 'https://api.github.com',
   gitlab: 'https://gitlab.com/api/v4',
-  bitbucket: 'https://api.bitbucket.org/2.0'
-}
+  bitbucket: 'https://api.bitbucket.org/2.0',
+};
 
 const api = {
   buildRequest(req: ApiRequest) {
-    return req
-  }
-}
+    return req;
+  },
+};
 
 function constructUrl(url: string, params?: ParamObject) {
   if (params) {
-    const paramList = []
+    const paramList = [];
     for (const key in params) {
-      paramList.push(`${key}=${encodeURIComponent(params[key])}`)
+      paramList.push(`${key}=${encodeURIComponent(params[key])}`);
     }
     if (paramList.length) {
-      url += `?${paramList.join('&')}`
+      url += `?${paramList.join('&')}`;
     }
   }
-  return url
+  return url;
 }
 
 async function constructRequestHeaders(headerConfig: HeaderConfig) {
-  const { token, headers } = headerConfig
-  const baseHeaders: HeaderObj = {'Content-Type': 'application/json; charset=utf-8', ...headers}
+  const { token, headers } = headerConfig;
+  const baseHeaders: HeaderObj = { 'Content-Type': 'application/json; charset=utf-8', ...headers };
   if (token) {
-    baseHeaders['Authorization'] = `token ${token}`
+    baseHeaders['Authorization'] = `token ${token}`;
```

Comment:
Hi @bytrangle, do you by any chance remember, why this is using `token` instead of bearer here?
This seems to be causing an issue with Gitlab logins, since this PR was merged in April: https://github.com/decaporg/decap-cms/issues/7172#issuecomment-2184138768

I'm not sure about Github/Bitbucket, but it seems at least for Gitlab this line should be like this:

```suggestion
    baseHeaders['Authorization'] = `Bearer ${token}`;
```

---

For reference:  This line was originally introduced in 3edc3cef1eeb905af25010e50524925c2748603e, this comment is on a2add4a4954aafa420f217af1d68867cc786e33a, which was later squashed to c91a70fbea7d6a4d413fc00cdd4a49b24fa97a16

Reply:
I don't remember. This corresponds to Gitlab docs 13.0, which is no longer available. Please feel free to make a new PR to correct it.

Reply:
Thanks anyway, I now created #7242 for this change.

Reply:
Nice work @floscher

I cloned the decapCMS repo locally yesterday to double the fix.

After adding the fix and rebuilding decapCMS, I no longer got the previous errir message.

However, I then still got an "API_ERROR: Branch not found" message, implying that the code this patch was meant to fix doesn't do it's job on gitlab anyway..

Only after hard-coding "branch: main" into my config.yml, was I then able to get on with things.